pyaedt/filtersolutions_core/dll_interface.py

```diff
@@ -0,0 +1,154 @@
+from typing import Callable
+
+
+class DllInterface:
+    """Interfaces with the FilterSolutions C++ API DLL."""
+
+    def __init__(self, show_gui=True):
+        self._init_dll_path()
+        self._init_dll(show_gui)
+
+    def restore_defaults(self):
+        """Restore the state of the API, including all options and values, to the initial startup state."""
+        status = self._dll.startApplication(self.show_gui)
+        self.raise_error(status)
+
+    def _init_dll_path(self):
+        """Set DLL path and print to screen the status of DLL access."""
+        relative_path = "../../../build_output/64Release/nuhertz/FilterSolutionsAPI.dll"
+        self.dll_path = os.path.join(os.path.dirname(__file__), relative_path)
+        if not os.path.isfile(self.dll_path):
+            self.dll_path = os.path.join(
+                os.environ["ANSYSEM_ROOT242"], "nuhertz/FilterSolutionsAPI.dll"
+            )  # pragma: no cover
+        print("DLL Path:", self.dll_path)
+        if not os.path.isfile(self.dll_path):
+            raise RuntimeError(f"The FilterSolutions API DLL was not found at {self.dll_path}")  # pragma: no cover
+
+    def _init_dll(self, show_gui):
+        """Load DLL and initiate application parameters to default values."""
+
+        self._dll = ctypes.cdll.LoadLibrary(self.dll_path)
+        self._define_dll_functions()
+        self.show_gui = show_gui
+        if show_gui:  # pragma: no cover
+            self._app_thread = threading.Thread(target=self._app_thread_task)
+            self._app_thread.start()
+            # TODO: Need some way to confirm that the GUI has completed initialization,
+            # otherwise some subsequent API calls will fail. For now, sleep a few seconds.
+            time.sleep(5)
+        else:
+            status = self._dll.startApplication(False)
+            self.raise_error(status)
+
+        print("DLL Loaded:", self.api_version())
+        print("API Ready")
+        print("")
+
+    def _app_thread_task(self):  # pragma: no cover
+        """Print the status of running application thread."""
+        print("Starting Application::Run thread")
+        status = self._dll.startApplication(self.show_gui)
+        self.raise_error(status)
+
+    def _define_dll_functions(self):
+        """Define DLL function."""
+        self._dll.getVersion.argtypes = [ctypes.c_char_p, ctypes.c_int]
+        self._dll.getVersion.restype = ctypes.c_int
+
+    def get_string(self, dll_function: Callable, max_size=100) -> str:
+        """
+        Call a DLL function that returns a string.
+
+        Parameters
+        ----------
+        dll_function: Callable
+            DLL function to be called. It must be a function that returns a string.
+        max_size: int
+            The maximum number of string characters to return. This will be used for the string buffer size.
+
+        Returns
+        -------
+        str
+            The requested string. If 'status' , Error message is returned.
```

Comment:
```suggestion
            Requested string. If '`status`', an error message is returned.
```
I'm not sure what the second sentence mans.